homeassistant/components/lock/__init__.py

```diff
@@ -87,40 +87,36 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     return True
 
 
-async def _async_lock(entity: LockEntity, service_call: ServiceCall) -> None:
-    """Lock the lock."""
-    code: str = service_call.data.get(
-        ATTR_CODE, entity._lock_option_default_code  # pylint: disable=protected-access
-    )
+@callback
+def _add_default_code(entity: LockEntity, service_call: ServiceCall) -> dict[Any, Any]:
+    code: str = service_call.data.get(ATTR_CODE, "")
+    if not code:
+        code = entity._lock_option_default_code  # pylint: disable=protected-access
```

Comment:
```suggestion
    code: str = service_call.data.get(
        ATTR_CODE, entity._lock_option_default_code  # pylint: disable=protected-access
    )
```
I think we can leave this as it was?